src/store/domains/todo/select.ts

```diff
@@ -1,35 +1,33 @@
 import { createSelector } from 'reselect';
-import { IState } from '../todo/interfaces';
+import { IStateToProps } from '../../../types';
 import { VisabilityList } from '../../../consts';
-import { getObjValues } from '../../../utils';
-
-interface IStateToProps {
-  todos: IState;
-  visabilityFilter: string;
-}
 
 export const selectTodos = (state: IStateToProps) => {
-  console.log(state.todos);
-  return getObjValues(state.todos.todoById);
+  return state.todos.todoById;
 };
 
 export const selectVisabilityFilter = (state: IStateToProps) => {
   return state.visabilityFilter;
 };
 
+export const selectVisableTodos = (state: IStateToProps) => {
+  return state.filteredTodos;
+};
+
 export const selectFilteredTodos = createSelector(
   selectTodos,
+  selectVisableTodos,
   selectVisabilityFilter,
-  (todos, filter) => {
+  (todos, visableTodos, filter) => {
     switch (filter) {
       case VisabilityList.ALL:
-        return todos;
+        return visableTodos.ALL.map(id => todos[id]);
```

Comment:
Here should be just visableTodos[currentFilter].map(id => todos[id]);
currentFilter should be taken from reducer